### ISSUE TITLE
CORERULES-9645 - is_column_of_iterables method only checks first row

### ISSUE THREAD

**Issue by ASL-rmarshall** [cdisc-org/cdisc-rules-engine#1017]
**Links to related JIRA Tickets**
- https://jira.cdisc.org/projects/CORERULES/issues/CORERULES-9645

**Rule Information**
- **Standard**: USDM
- **Rule ID**: DDF00102
- **Rule Description**: A scheduled activity instance must only reference a timeline exit that is defined within the same schedule timeline as the scheduled activity instance.

**Describe the bug**
As shown in "Request.txt" attached below, this rule was originally implemented by using the `distinct` operator to retrieve a list of the distinct schedule timeline exit `id` values grouped by `parent_id`, which was merged onto the check dataset.  Error records were reported when the value on the `timelineExitId` variable in the check dataset `is_not_contained_by` the unique list of schedule timeline exit `id` values.  However, the rule failed with an error:
> Cannot perform 'and_' with a dtyped [bool] array and scalar of type [NoneType]

This was happening because the grouped list of unique values was only populated on records with matching group values - it was `None` for other records.  The `is_column_of_iterables` method is used to check if the comparison column contains iterables to determine the method used to do the "contains" comparison. However, as `is_column_of_iterables` only checks the first row, the comparison method used when the first row contains an iterable fails for any other row that contains `None`.

The same issue occurs if the `does_not_contain` operator is used.

Note that this rule has now been rewritten to use a table join instead, so this is no longer a rule-blocking issue.

**Error returned from Rule Engine**
```json
    "ScheduledActivityInstance": [
        {
            "executionStatus": "execution_error",
            "dataset": "ScheduledActivityInstance.xpt",
            "domain": "",
            "variables": [],
            "message": "rule execution error",
            "errors": [
                {
                    "dataset": "ScheduledActivityInstance.xpt",
                    "error": "An unknown exception has occurred",
                    "message": "Cannot perform 'and_' with a dtyped [bool] array and scalar of type [NoneType]"
                }
            ]
        }
    ],
```

**Expected behavior**
The `is_not_contained_by` and `does_not_contain` operators should run without error when comparing a variable value against a grouped list of unique values generated by the `distinct` operator.

**Test data**

[Request.txt](https://github.com/user-attachments/files/18766543/Request.txt)

Test data is also available in Excel format in the USDM > DDF00102 > positive > 01 > data folder of the CORE unitTesting SharePoint.

**Comment by ASL-rmarshall** [comment]
Reviewed - this is a general bug, so review and a fix is still needed.

**Comment by ASL-rmarshall** [comment]
The `is_column_of_iterables` method has been rewritten to check all rows, but it is still not returning the correct result. At the moment, it only returns `True` if every value in the column is an iterable. However, in the scenario described in the issue above - where the result of a grouped `distinct` operation that does not merge back onto every row - the column checked by `is_column_of_iterables` might contain null values.  I think `is_column_of_iterables` needs to check only the non-null values and return `True` if they are all iterables, e.g., something like:
```
    def is_column_of_iterables(self, column):
        return self.value.is_series(column) and all(            
            isinstance(val, (list, set))
            for val in column if
            not (val is None
            or (isinstance(val, float) and pd.isna(val)))
        )
```
This issue potentially affects any rule that includes a grouped `distinct` operation operator whose result is the target or comparator for any operator that uses `is_column_of_iterables`:
- `contains`
- `does_not_contain`
- `contains_case_insensitive`
- `does_not_contain_case_insensitive`
- `is_contained_by`
- `is_not_contained_by`
- `is_contained_by_case_insensitive`
- `is_not_contained_by_case_insensitive`
- `contains_all`
- `not_contains_all`
- `prefix_is_contained_by`
- `prefix_is_not_contained_by`
- `suffix_is_contained_by`
- `suffix_is_not_contained_by`

There are currently 9 published USDM rules that are potentially impacted by this issue:
- DDF00024
- DDF00028
- DDF00029
- DDF00046
- DDF00047
- DDF00049
- DDF00115
- DDF00151
- DDF00152

Of these, we have already seen incorrect results returned by DDF00152 when run using this data:
[unit-test-coreid-DDF00152-positive-tst.xlsx](https://github.com/user-attachments/files/23927217/unit-test-coreid-DDF00152-positive-tst.xlsx)
